swarm/network/kademlia.go

```diff
@@ -725,7 +726,11 @@ func (k *Kademlia) getUnsaturatedBins(peersPerBin []int, depth int) []int {
 	})
 
 	log.Trace("list of unsaturated bins", "unsaturatedBins", unsaturatedBins)
-	return unsaturatedBins
+	// early check for depth
```

Comment:
this should come BEFORE, if it is false you dont need to call the EachBin thing.